test/games/fusion/gui/preset_settings/test_fusion_game_patches.py

```diff
@@ -0,0 +1,34 @@
+from __future__ import annotations
+
+import dataclasses
+import uuid
+from unittest.mock import MagicMock
+
+from PySide6 import QtCore
+
+from randovania.game_description import default_database
+from randovania.games.fusion.gui.preset_settings.fusion_patches_tab import PresetFusionPatches
+from randovania.games.fusion.layout.fusion_configuration import FusionConfiguration
+from randovania.games.game import RandovaniaGame
+from randovania.interface_common.preset_editor import PresetEditor
+
+
+def test_on_preset_changed(skip_qtbot, preset_manager):
+    # Setup
+    game = RandovaniaGame.FUSION
+    options = MagicMock()
+
+    base = preset_manager.default_preset_for_game(game).get_preset()
+    preset = dataclasses.replace(base, uuid=uuid.UUID("bb84005f-b0f1-4e7e-8057-f4275ff4b2a3"))
+    editor = PresetEditor(preset, options)
+    window = PresetFusionPatches(editor, default_database.game_description_for(game), MagicMock())
+    skip_qtbot.addWidget(window)
+
+    # Run
+    window.on_preset_changed(editor.create_custom_preset_with())
+    skip_qtbot.mouseClick(window.instant_transitions_check, QtCore.Qt.MouseButton.LeftButton)
+
+    # Assert
+    final_preset = editor.create_custom_preset_with()
+    assert isinstance(final_preset.configuration, FusionConfiguration)
+    assert final_preset.configuration.instant_transitions is True
```

Comment:
this would be nicer if it checks all checkboxes.
You can get all checkboxes via findChild. Then get the _FIELDS attribute, compare that a) both are the same length and b) enabling all leads to properties getting enabled, and disabling all, leads to all properties getting disabled.